Microsoft.Toolkit.Uwp.UI.Controls/TextBoxMask/TextBoxMask.cs

```diff
@@ -119,10 +120,38 @@ private static void Textbox_Loaded(object sender, RoutedEventArgs e)
             textbox.TextChanging += Textbox_TextChanging;
             textbox.SelectionChanged += Textbox_SelectionChanged;
             textbox.Paste += Textbox_Paste;
+            textbox.GotFocus += Textbox_GotFocus;
             textbox.SetValue(OldTextProperty, textbox.Text);
             textbox.SelectionStart = 0;
         }
 
+        private static void Textbox_GotFocus(object sender, RoutedEventArgs e)
+        {
+            var textbox = sender as TextBox;
+            var mask = textbox?.GetValue(MaskProperty) as string;
+            var placeHolderValue = textbox?.GetValue(PlaceHolderProperty) as string;
+            var representationDictionary = textbox?.GetValue(RepresentationDictionaryProperty) as Dictionary<char, string>;
+            if (string.IsNullOrWhiteSpace(mask) ||
+                representationDictionary == null ||
+                string.IsNullOrEmpty(placeHolderValue))
+            {
+                return;
+            }
+
+            var placeHolder = placeHolderValue[0];
+            var displayText = mask;
+            foreach (var key in representationDictionary.Keys)
+            {
+                displayText = displayText.Replace(key, placeHolder);
+            }
+
+            // if the textbox got focus and the textbox is empty (contains only mask) set the textbox cursor at the beginning to simulate normal TextBox behavior if it is empty.
+            if (string.Equals(textbox.Text, displayText))
```

Comment:
Should probably just always put the cursor to the end of where the user has entered text. If they have entered no text, it's at 0. If they have entered 3 of 10 characters (eg: phone number) then it's at 3.